docs/how-to/integrate_sdcore_with_observability.md

```diff
@@ -1,79 +1,96 @@
+### Pre-requisites
 
-```bash
-juju add-model cos 
-juju deploy cos-lite --trust
-```
+- A Kubernetes cluster capable of handling the load from both SD-Core and COS
+- [Charmed 5G Terraform modules][Charmed 5G Terraform modules] Git repository cloned onto the Juju host machine
 
-## Deploy the `cos-configuration-k8s` charm
+### Including COS integration in the SD-Core deployment
 
-Deploy the `cos-configuration-k8s` charm with the following SD-Core COS configuration:
+Inside the directory of a desired SD-Core Terraform module, create `variables.tfvars` file and add following line(s) to it:
 
 ```console
-juju deploy cos-configuration-k8s \
-  --config git_repo=https://github.com/canonical/sdcore-cos-configuration \
-  --config git_branch=main \
-  --config git_depth=1 \
-  --config grafana_dashboards_path=grafana_dashboards/sdcore/
+deploy_cos = true
+cos_model_name = "YOUR_CUSTOM_COS_MODEL_NAME" (Optional. Defaults to `cos-lite`.)
+cos_configuration_config = {} (Optional. Allows customization of the `COS Configuration` application.)
 ```
 
-Integrate it with Grafana:
-
-```console
-juju integrate cos-configuration-k8s grafana
+```{note}
+If you have already created the `.tfvars` file, to customize the deployment of SD-Core, you should edit the existing file rather than create a new one.
 ```
 
-## Integrate Grafana Agent with Prometheus
+Proceed with the deployment.
 
-We will create a cross model integration between Grafana Agent (in the SD-Core model) and Prometheus (in the `cos` model).
+(option-2)=
+## Integrating COS with an existing SD-Core deployment
 
-First, offer the following integrations from Prometheus and Loki for use in other models:
+This option allows deploying COS and integrating it with SD-Core as a Day 2 operation.
 
-```bash
-juju offer cos.prometheus:receive-remote-write
-juju offer cos.loki:logging
-```
+### Pre-requisites
 
-Then, consume the integrations from the SD-Core model:
+- A Kubernetes cluster capable of handling the load from both SD-Core and COS
+- Any [Charmed 5G Terraform module][Charmed 5G Terraform modules] deployed
 
-```bash
-juju switch <SD-Core model>
-juju consume cos.prometheus
-juju consume cos.loki
+### Adding COS to an existing SD-Core deployment
+
+Go to a directory from which SD-Core was deployed (the one containing Terraform's `.tfstate` file).
+Edit the `.tfvars` and add following line(s) to it:
+
+```console
+deploy_cos = true
+cos_model_name = "YOUR_CUSTOM_COS_MODEL_NAME" (Optional. Defaults to `cos-lite`.)
+cos_configuration_config = {} (Optional. Allows customization of the `COS Configuration` application.)
 ```
 
-Integrate `grafana-agent-k8s` with `prometheus` and `loki`:
+Apply the changes:
 
-```bash
-juju integrate prometheus:receive-remote-write grafana-agent-k8s:send-remote-write
-juju integrate loki:logging grafana-agent-k8s:logging-consumer
+```console
+terraform apply -var-file="<YOUR_TFVARS_FILE>" -auto-approve
 ```
```

Comment:
Here, I think we need add some comments to wait till all applications become Active/Idle state. Because the next stage directly starts with getting grafana admin password. If we the users immediately goes to the next stage, the action will fail.

Reply:
Added.